lib/OpenQA/Worker.pm

```diff
@@ -687,6 +693,11 @@ sub _handle_job_status_changed {
             $self->_clean_pool_directory;
         }
 
+        # update the general worker availability (e.g. we might detect here that QEMU from the last run
+        # hasn't been terminated yet)
+        # incomplete subsequent jobs in the queue if it turns out the worker is generally broken
+        return $self->_accept_or_skip_next_job_in_queue('worker broken') unless $self->check_availability;
```

Comment:
I might've set `$reason = 'worker broken' here rather than duplicating the method call, since I'd find it slightly clearer.